synapsebridgehelpers/export_tables.py

```diff
@@ -0,0 +1,342 @@
+
+
+def export_tables(syn, table_mapping, target_project=None, update=True,
+                  reference_col="recordId", copy_file_handles=True, **kwargs):
+    """Export records from one Synapse table to another. Or transfer tables
+    to completely new tables in a separate project.
+
+    Parameters
+    ----------
+    syn : synapseclient.Synapse
+    table_mapping : dict, list, or str
+        If exporting records of one or more tables to other, preexisting tables,
+        table_mapping is a dictionary containing Synapse ID key/value mappings
+        from source to target tables. If exporting table records to not yet
+        created tables in a seperate project, table_mapping can be a list or
+        string.
+    target_project : str
+        If exporting table records to not yet created tables in a seperate
+        project, specify the target projects Synapse ID here.
+    update : bool, defaults to True
+        When exporting records of one or more tables to other, preexisting
+        tables, whether to append new records to these tables or completely
+        overwrite the table records.
+    referenceCol : str
+        If `update` is True, use this column as the table index to determine
+        which records are already present in the target table.
+    copy_file_handles : bool, defaults to True
+        Whether to copy the file handles from the source table to the target
+        table. If you are not the creator of these file handles, this must
+        be set to True if you want to a column containing file handles in your
+        target table.
+    **kwargs
+        Additional arguments to pass to synapsebridgehelpers.query_across_tables
+
+    Returns
+    -------
+    """
+    results = {}
+    if isinstance(table_mapping, (list, str)): # export to brand new tables
+        if target_project is None:
+            raise TypeError("If passing a list to table_mapping, "
+                            "target_project must be set.")
+        source_tables = synapsebridgehelpers.query_across_tables(
+                syn, tables=table_mapping, as_data_frame=True, **kwargs)
+        if isinstance(table_mapping, str):
+            source_table_iter = [(table_mapping, source_tables[0])]
+        else:
+            source_table_iter = zip(table_mapping, source_tables)
+        for source_id, source_table in source_table_iter:
+            source_table_info = syn.get(source_id)
+            source_table_cols = list(syn.getTableColumns(source_id))
+            if copy_file_handles:
+                source_table = replace_file_handles(
+                        syn,
+                        df = source_table,
+                        source_table_id = source_id,
+                        source_table_cols = source_table_cols)
+            sanitized_source_table = sanitize_table(
+                    syn,
+                    records = source_table,
+                    cols = source_table_cols)
+            target_table_schema = sc.Schema(
```

Comment:
sanitize is not very descriptive of what you are doing.

Reply:
`format_table`? I can't think of a succinct way to be more explicit without naming the function something like `modify_column_types_for_synapse_store`.

"sanitize" is descriptive in the sense that we are taking dirty integer values [123.0, 456.0, ...] and turning them into clean integer values [123, 456], plus it leaves the door open for future pandas DataFrame values clean up that doesn't involve taking the ".0" off of integers.

Reply:
change to sanitize_dataframe